lib/scope.js

```diff
@@ -650,9 +650,34 @@ module.exports = {
     }
   },  // Ends scope.setLanguage()
 
-  normalizeTable: async function ( scope, { var_data, from_story_table=true }) {
-    /* Return data derived from cucumber variable-setting data.
-     * Table must have headers. Arrays are not supported. */
+  normalizeTable: async function ( scope, { var_data=null, raw_var_data=null, from_story_table=true }) {
+    /* Return data derived from cucumber variable-setting data, from either an array or object format */
+
+    if (!var_data && ! raw_var_data) {
+      let no_data_msg = `ALKiln Internal error: normalizeTable called without var_data or raw_var_data`;
+      await scope.addToReport( scope, { type: `error`, value: no_data_msg });
+      throw Error(bad_table_msg);
+    }
+
+    // Always use the raw var data if present; it has the full first row
+    if (raw_var_data) {
+      var_data = raw_var_data.hashes();
+      if (!('var' in var_data[0]) || !('value' in var_data[0])) {
+        let raw_array = raw_var_data.raw();
+        var_data = [];
+        for (let row_raw of raw_array) {
+          if (row_raw.length < 2) {
+            let bad_table_msg = `Your Story Table definition is bad: the first row should be "| var | value | trigger |", but it's actually ${ raw_array[0] }`;
```

Comment:
This looks like the old "bad table msg" that was just for when the first row had the wrong column headings. It looks like this could happen if any row had only one column and that seems like it'd be a different error message.
```suggestion
            let bad_table_msg = `Your Story Table definition needs to be changed. A row in your table only has 1 column when it should have at least 2.`;
```

Reply:
My way of designing this is that it should be very clear what you're encouraged to do, but we should be flexible if someone made a typo, or forgot their header row. We don't have to leave this feature officially undocumented (i.e. not on the docs page, but still in changelog, etc.), but I do think that all of the error messages should assume that you're doing the standard way of doing things, like including a header row with three columns.

Reply:
That's not what the if statements tests for, though, right? It would lead to inconsistent behavior from the point of view of the developer as they could have other tests that don't have a header row, but are working. That might make it hard for them to track down the actual problem.

I'm not totally against it, but it at least should have a note as to why the message doesn't reflect the logic test that triggers the message.

As a separate note, I think "is bad" is an unpleasant way for someone to experience this error.

Reply:
Maybe something like:
```
`... The best practice is to make your table 3 columns wide with a first row in this format: "| var...
```

Reply:
Agreed that "is bad" is unpleasant, I had already removed that. And I had landed on something similar to your last suggestion (I was gonna use "we don't have enough information to know how to use your Story Table"), but I like your "best practice" wording better, should be in now.